packages/flutter_driver/lib/src/extension.dart

```diff
@@ -20,19 +22,26 @@ import 'gesture.dart';
 import 'health.dart';
 import 'message.dart';
 import 'render_tree.dart';
+import 'request_data.dart';
 import 'semantics.dart';
 
 const String _extensionMethodName = 'driver';
 const String _extensionMethod = 'ext.flutter.$_extensionMethodName';
 
-class _DriverBinding extends WidgetsFlutterBinding { // TODO(ianh): refactor so we're not extending a concrete binding
+typedef String DataHandler(String message);
```

Comment:
If this returned `Future<String>` it could be used for more use-cases (such as being able to wait for data to become available). You shouldn't lose anything in generality given that `_requestData` below and the network separation by its nature are already asynchronous.